### ISSUE TITLE
V5.44 DriveItem track changes - missing functionality

### ISSUE THREAD

**Issue by Yosif-Smint** [microsoftgraph/msgraph-sdk-dotnet#2383]
Hello,

I am migrating Microsoft.Graph version `4.25` to `5.44`

One of the challenges that I am facing is with `DriveItem` track changes functionality

According to the documentation [here](https://learn.microsoft.com/en-us/graph/api/driveitem-delta?view=graph-rest-1.0&tabs=csharp)  

`GET /sites/{siteId}/drive/root/delta` should be supported

When I run a raw query against the URL below, the expected response is returned

```
https://graph.microsoft.com/v1.0/sites/*.sharepoint.com%2Cbedcabf9-6.../drive/root/delta(token='NDsIMj...')?$top=10
```

`DeltaWithTokenRequestBuilder.cs` does not have a constructor or a method for `urlTemplate` override

```
{+baseurl}/sites/{site%2Did}/drive/root/delta(token='{token}'){?%24count,%24expand,%24filter,%24orderby,%24search,%24select,%24skip,%24top}
```

This API also works for a specific drive ID, but is not documented
```
https://graph.microsoft.com/v1.0/sites/*.sharepoint.com%2Cbedcabf9-6.../drives/b!-avcv.../root/delta(token='NDsIMj...')?$top=10
```

Please consider supporting it as well

```
{+baseurl}/sites/{site%2Did}/drives/{drive%2Did}/root/delta(token='{token}'){?%24count,%24expand,%24filter,%24orderby,%24search,%24select,%24skip,%24top}
```

I tried all sorts of things like `CustomDriveItemItemRequestBuilder` `DeltaWithToken`, `WithUrl` and nothing worked

The solution right now is to do it manually

```C#
var urlTemplate = "{+baseurl}/sites/{site%2Did}/drive/root/delta(token='{token}'){?%24count,%24expand,%24filter,%24orderby,%24search,%24select,%24skip,%24top}";

Dictionary<string, object> pathParameters = new()
{
    { "baseurl", string.Format(Constants.Url.GraphBaseUrlFormatString, "v1.0") },
    { "site%2Did", SiteId },
    { "%24top", "50" },
    { "token", "latest" }
};

if (!string.IsNullOrEmpty(_siteDriveId))
{
    urlTemplate = "{+baseurl}/sites/{site%2Did}/drives/{drive%2Did}/root/delta(token='{token}'){?%24count,%24expand,%24filter,%24orderby,%24search,%24select,%24skip,%24top}";

    pathParameters.Add("drive%2Did", _siteDriveId);
}

var requestInformation = new RequestInformation(Method.GET, urlTemplate, pathParameters);

var deltaWithTokenRequestBuilder = new DeltaWithTokenRequestBuilder(requestInformation.URI.ToString(), _graphServiceClient.RequestAdapter);

var deltaWithTokenGetResponse = await deltaWithTokenRequestBuilder.GetAsDeltaWithTokenGetResponseAsync();

// ...
```

Best Regards,
Yosif

**Comment by andrueastman** [comment]
Thanks for raising this @Yosif-Smint 

This is addressed in the documentation at https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/upgrade-to-v5.md#drive-item-paths

We'll close this for now as adding the extra segments to the SDK won't be supported for now due to the path increases.

**Comment by bretthacker** [comment]
The link in the last comment is a 404. You've closed this as completed but this SDK functionality is completely broken. We are reduced to writing our own REST wrappers, which obviates the need for an SDK. This ill-advised move to Kiota is one of the biggest disasters I've encountered as a consumer of Graph. I moved from 3.x to 5.x and am not looking at moving back to 4.x, or throwing the whole pile of crap out and writing my own library, which given the scope would essentially be a new business for me.

v5 is a disaster.
